Corona-Warn-App/src/main/java/de/rki/coronawarnapp/http/playbook/PlaybookImpl.kt

```diff
@@ -0,0 +1,143 @@
+
+        // fake registration
+        ignoreExceptions { webRequestBuilder.asyncFakeGetRegistrationToken() }
+
+        // fake submission
+        ignoreExceptions { webRequestBuilder.asyncFakeSubmitKeysToServer() }
+
+        coroutineScope.launch { followUpPlaybooks() }
+
+        return testResult?.let { TestResult.fromInt(it) }
+            ?: propagateException(exception)
+    }
+
+    override suspend fun submission(
+        registrationToken: String,
+        keys: List<KeyExportFormat.TemporaryExposureKey>
+    ) {
+        Timber.i("[$uid] New Submission Playbook")
+
+        // real auth code
+        val (authCode, exception) = executeCapturingExceptions {
+            webRequestBuilder.asyncGetTan(
+                registrationToken
+            )
+        }
+
+        // fake registration
+        ignoreExceptions { webRequestBuilder.asyncFakeGetRegistrationToken() }
+
+        // real submission
+        if (authCode != null) {
+            webRequestBuilder.asyncSubmitKeysToServer(authCode, keys)
+            coroutineScope.launch { followUpPlaybooks() }
+        } else {
+            webRequestBuilder.asyncFakeSubmitKeysToServer()
+            coroutineScope.launch { followUpPlaybooks() }
+            propagateException(exception)
+        }
+    }
+
+    private suspend fun dummy(launchFollowUp: Boolean) {
+        // fake registration
+        ignoreExceptions { webRequestBuilder.asyncFakeGetRegistrationToken() }
+
+        // fake test result
+        ignoreExceptions { webRequestBuilder.asyncFakeGetTestResult() }
+
+        // fake submission
+        ignoreExceptions { webRequestBuilder.asyncFakeSubmitKeysToServer() }
+
+        if (launchFollowUp)
+            coroutineScope.launch { followUpPlaybooks() }
+    }
+
+    override suspend fun dummy() = dummy(true)
+
+    private suspend fun followUpPlaybooks() {
+        val runsToExecute = Random.nextInt(
+            SubmissionConstants.minNumberOfSequentialPlaybooks,
+            SubmissionConstants.maxNumberOfSequentialPlaybooks
+        )
+        Timber.i("[$uid] Follow Up: launching $runsToExecute follow up playbooks")
+
+        repeat(runsToExecute) {
+            val executionDelay = Random.nextInt(
+                SubmissionConstants.minDelayBetweenSequentialPlaybooks,
+                SubmissionConstants.maxDelayBetweenSequentialPlaybooks
+            )
+            Timber.i("[$uid] Follow Up: (${it + 1}/$runsToExecute) waiting $executionDelay[s]...")
+            delay(TimeUnit.SECONDS.toMillis(executionDelay.toLong()))
+
+            dummy(false)
+        }
+        Timber.i("[$uid] Follow Up: finished")
+    }
+
+    private suspend fun ignoreExceptions(body: suspend () -> Unit) {
+        try {
+            body.invoke()
+        } catch (e: Exception) {
+            Timber.d(e, "Ignoring dummy request exception")
+        }
+    }
+
+    private suspend fun <T> executeCapturingExceptions(body: suspend () -> T): Pair<T?, Exception?> {
```

Comment:
Maybe a simple throw can be enough here depending on the use case.

Reply:
We need to ensure that following fake requests of the playbook are still executed even when the real request fails